app/src/main/java/com/lightricks/feedexercise/database/FeedDatabase.kt

```diff
@@ -1,5 +1,31 @@
 package com.lightricks.feedexercise.database
 
+import android.content.Context
+import androidx.room.Database
+import androidx.room.Room
+import androidx.room.RoomDatabase
+
 /**
  * todo: add the abstract class that extents RoomDatabase here
  */
+
+@Database(entities = [FeedItemEntity ::class], version = 1)
+abstract class FeedDatabase : RoomDatabase() {
+
+    abstract val feedDao : FeedDao
+
+    companion object {
+        @Volatile private var INSTANCE: FeedDatabase? = null
+
+        @Synchronized
+        fun getInstance(context : Context): FeedDatabase {
```

Comment:
if you put the synchronization on all the method then you lost the benefit of volatile.